pkg/crud_test.go

```diff
@@ -0,0 +1,133 @@
+package pkg
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func GetToolStruct() Tool {
+	tool := Tool{
+		Name:    "dnsx",
+		Repo:    "dnsx",
+		Version: "1.1.1",
+		Assets: map[string]string{
+			"dnsx_1.1.1_checksums.txt":     "79344865",
+			"dnsx_1.1.1_linux_386.zip":     "79344862",
+			"dnsx_1.1.1_linux_amd64.zip":   "79344859",
+			"dnsx_1.1.1_linux_arm64.zip":   "79344852",
+			"dnsx_1.1.1_linux_armv6.zip":   "79344864",
+			"dnsx_1.1.1_macOS_amd64.zip":   "79344851",
+			"dnsx_1.1.1_macOS_arm64.zip":   "79344856",
+			"dnsx_1.1.1_windows_386.zip":   "79344855",
+			"dnsx_1.1.1_windows_amd64.zip": "79344857",
+		},
+	}
+
+	return tool
+}
+
+func TestInstallTool(t *testing.T) {
+	token := os.Getenv("GITHUB_TOKEN")
```

Comment:
The GH token is already picked up internally at https://github.com/projectdiscovery/pdtm/blob/4b34734b4141f2351dd423b88b0bbc9078b5a8ba/pkg/gh.go#L11 - I think the test should be executed always, if it fails hard because of rate-limit, then the user should configure a token

Reply:
Added check on rate limit on Ghclient:

```golang
rateLimit, _, err := githubClient.RateLimits(context.Background())
	if err != nil {
		gologger.Error().Msgf("error while getting rate limits %s ", err.Error())

	}
	if rateLimit.Core.Remaining <= 0 {
		gologger.Error().Msgf("error for remaining request per hour: %s", err.Error())

	}
```